File: 0187-repeated-dna-sequences/0187-repeated-dna-sequences.py
class Solution:
    def findRepeatedDnaSequences(self, s: str) -> List[str]:
        seen_hash = set()
        result = set()
        
        for ed in range(len(s) - 9):
            current = s[ed : ed + 10]
            if current in seen_hash:
                result.add(current)
            seen_hash.add(current)
            
        return list(result)
        

File: 0187-repeated-dna-sequences/0187-repeated-dna-sequences.java
import java.util.*;

public class Solution {
    public List<String> findRepeatedDnaSequences(String s) {
        Set<String> L1 = new HashSet<>();
        Set<String> L2 = new HashSet<>();
        int l = s.length();
        if (l <= 10)
            return new ArrayList<>(L1);
        for (int i = 0; i <= l - 10; i++) {
            String stri = s.substring(i, i + 10);
            if (L1.contains(stri) && !L2.contains(stri)) {
                L2.add(stri);
            } else if (!L1.contains(stri)) {
                L1.add(stri);
            }
        }
        
        return new ArrayList<>(L2);
    }
}
